### PR TITLE
Extending tagged_file_sink file naming options.

### PR DESCRIPTION
**Pull request by aiicore**
- Introducing target directory configurable option.
- Introducing two naming styles: legacy and non legacy.

Legacy:
 - Works as before.

Non legacy:
 - Introduces configurable filename prefix.
 - Naming convention has chaned to:
  ` <prefix>_%Y-%m-%d_%H_%M_%S.%f.dat`

Signed-off-by: Igor Podoski <igor.podoski@gmail.com>

### CODE REVIEW

**Comment by aiicore** [comment]
Introduces feature from #2907,  inspired by #3010.

File: gr-blocks/lib/tagged_file_sink_impl.cc [hunk 1]
```diff
@@ -142,19 +156,43 @@ int tagged_file_sink_impl::work(int noutput_items,
...
+                        filename.setf(std::ios::fixed, std::ios::floatfield);
+                        filename.precision(8);
+                        filename << "file" << unique_id() << "_" << d_n << "_"
+                                 << d_timeval;
+                    } else {
+                        const boost::posix_time::ptime now =
+                            boost::posix_time::microsec_clock::local_time();
+                        static std::locale time_date(
+                            filename.getloc(),
+                            new boost::posix_time::time_facet("%Y-%m-%d_%H_%M_%S.%f"));
```
Comment (aiicore) [inline gr-blocks/lib/tagged_file_sink_impl.cc:169]: This precision will be sufficient, or should I add d_n or d_timeval also?

**Comment by aiicore** [comment]
@dkozel, did few improvements: Filename prefix is now required in non-legacy mode, also is related with state of legacy switch (GUI). Class declaration have more default values, so any old api that would call tagged_file_sink's make, should work as before.

**Comment by mormj** [comment]
@aiicore - were you able to get a CLA in place?

**Comment by aiicore** [comment]
@mormj not yet, the process still is ongoing.

**Comment by aiicore** [comment]
Can't get CLA.

**Pull request closed without merging**